hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -17,281 +17,82 @@
+    this.copy(prunedDataSchema = Some(prunedSchema))
+
+  protected override def timeline: HoodieTimeline = {
+    if (fullTableScan) {
+      metaClient.getCommitsAndCompactionTimeline
+    } else if (useStateTransitionTime) {
+      metaClient.getCommitsAndCompactionTimeline.findInstantsInRangeByStateTransitionTs(startTimestamp, endTimestamp)
     } else {
-      commitTimeline.findInstantsInRange(
-        optParams(DataSourceReadOptions.BEGIN_INSTANTTIME.key),
-        optParams.getOrElse(DataSourceReadOptions.END_INSTANTTIME.key(), lastInstant.getTimestamp))
+      metaClient.getCommitsAndCompactionTimeline.findInstantsInRange(startTimestamp, endTimestamp)
     }
   }
-  private val commitsToReturn = commitsTimelineToReturn.getInstantsAsStream.iterator().toList
-
-  // use schema from a file produced in the end/latest instant
 
-  val (usedSchema, internalSchema) = {
-    log.info("Inferring schema..")
-    val schemaResolver = new TableSchemaResolver(metaClient)
-    val iSchema : InternalSchema = if (!isSchemaEvolutionEnabledOnRead(optParams, sqlContext.sparkSession)) {
-      InternalSchema.getEmptyInternalSchema
-    } else if (useEndInstantSchema && !commitsToReturn.isEmpty) {
-      InternalSchemaCache.searchSchemaAndCache(commitsToReturn.last.getTimestamp.toLong, metaClient, hoodieTable.getConfig.getInternalSchemaCacheEnable)
-    } else {
-      schemaResolver.getTableInternalSchemaFromCommitMetadata.orElse(null)
-    }
-
-    val tableSchema = if (useEndInstantSchema && iSchema.isEmptySchema) {
-      if (commitsToReturn.isEmpty) schemaResolver.getTableAvroSchema(false) else
-        schemaResolver.getTableAvroSchema(commitsToReturn.last, false)
-    } else {
-      schemaResolver.getTableAvroSchema(false)
-    }
-    if (tableSchema.getType == Schema.Type.NULL) {
-      // if there is only one commit in the table and is an empty commit without schema, return empty RDD here
-      (StructType(Nil), InternalSchema.getEmptyInternalSchema)
-    } else {
-      val dataSchema = AvroConversionUtils.convertAvroSchemaToStructType(tableSchema)
-      if (iSchema != null && !iSchema.isEmptySchema) {
-        // if internalSchema is ready, dataSchema will contains skeletonSchema
-        (dataSchema, iSchema)
-      } else {
-        (StructType(skeletonSchema.fields ++ dataSchema.fields), InternalSchema.getEmptyInternalSchema)
-      }
-    }
+  protected override def composeRDD(fileSplits: Seq[HoodieBaseFileSplit],
+                                    tableSchema: HoodieTableSchema,
+                                    requiredSchema: HoodieTableSchema,
+                                    requestedColumns: Array[String],
+                                    filters: Array[Filter]): RDD[InternalRow] = {
+    super.composeRDD(fileSplits, tableSchema, requiredSchema, requestedColumns,
+      filters ++ incrementalSpanRecordFilters)
   }
 
-  private val filters = optParams.getOrElse(DataSourceReadOptions.PUSH_DOWN_INCR_FILTERS.key,
```

Comment:
Hey @bvaradar, it appears this is introduced in https://github.com/apache/hudi/pull/485, but I don't find any implementation in `MergeOnReadIncrementalRelation`, Do we overlook it or this configure is a legacy one and no longer required?